contracts/protocol/tokenization/APoRToken.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.6.12;
+
+import {AToken} from './AToken.sol';
+import {IAPoRToken} from '../../interfaces/IAPoRToken.sol';
+import {ILendingPool} from '../../interfaces/ILendingPool.sol';
+import {IChainlinkAggregatorV3} from '../../interfaces/IChainlinkAggregatorV3.sol';
+import {SafeMath} from '../../dependencies/openzeppelin/contracts/SafeMath.sol';
+import {IERC20Detailed} from '../../dependencies/openzeppelin/contracts/IERC20Detailed.sol';
+import {Errors} from '../libraries/helpers/Errors.sol';
+
+/**
+ * @title Aave's APoRToken (aToken with proof-of-reserves) Contract
+ * @notice AToken that checks reserves before minting
+ * @author Chainlink
+ */
+contract APoRToken is AToken, IAPoRToken {
+  using SafeMath for uint256;
+
+  uint256 public constant MAX_AGE = 7 days;
+  address public feed;
+  uint256 public heartbeat;
+
+  modifier onlyPoolAdmin {
+    require(
+      _msgSender() == ILendingPool(_pool).getAddressesProvider().getPoolAdmin(),
+      Errors.CALLER_NOT_POOL_ADMIN
+    );
+    _;
+  }
+
+  /**
+   * @dev This constructor only serves to set a default heartbeat.
+   *  Don't forget to use `initialize(...)` as you would with a regular AToken.
+   */
+  constructor() public {
+    heartbeat = MAX_AGE;
+  }
+
+  /**
+   * @notice Overriden mint function that checks the specified proof-of-reserves feed to
+   * ensure that the supply of the underlying assets is not greater than the reported
+   * reserves.
+   * @dev The proof-of-reserves check is bypassed if feed is not set.
+   * @param account The address to mint tokens to
+   * @param amount The amount of tokens to mint
+   */
+  function _mint(address account, uint256 amount) internal virtual override {
+    if (feed == address(0)) {
+      super._mint(account, amount);
+      return;
+    }
+
+    // Get latest proof-of-reserves from the feed
+    (, int256 answer, , uint256 updatedAt, ) = IChainlinkAggregatorV3(feed).latestRoundData();
+    require(answer > 0, Errors.AT_POR_INVALID_ANSWER);
+
+    // Check the answer is fresh enough (i.e., within the specified heartbeat)
+    uint256 heartbeat_ = heartbeat;
+    uint256 oldestAllowed = block.timestamp.sub(heartbeat_, Errors.AT_POR_INVALID_TIMESTAMP);
```

Comment:
Small gas optimization, we can skip storing `heartbeat` to memory here since we're only using it once in this function now.

Reply:
Good lookin out 👍  Updated.